ui/packages/catalog-ui-search/src/main/webapp/component/result-item/result-item.view.js

```diff
@@ -449,19 +450,27 @@ const ResultItemView = Marionette.LayoutView.extend({
         .get('associations.external') !== undefined
     )
   },
-  checkDisplayType: function() {
-    var displayType = user
-      .get('user')
-      .get('preferences')
-      .get('resultDisplay')
-    switch (displayType) {
-      case 'List':
+  checkDisplayType() {
+    const incomingDisplayType = getResultDisplayType()
+
+    switch (incomingDisplayType) {
+      case LIST_DISPLAY_TYPE:
         this.$el.removeClass('is-grid').addClass('is-list')
         break
-      case 'Grid':
+      case GRID_DISPLAY_TYPE:
         this.$el.addClass('is-grid').removeClass('is-list')
         break
     }
+
+    //Only re-render when going from `List` to `Grid`
+    if (
```

Comment:
We don't need to store off the `currentDisplayType`.  If we get a change event for resultDisplay we're guaranteed that it's going to differ.